linux/LibSensors.c

```diff
@@ -208,6 +212,33 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
          if (r != 0)
             continue;
 
+         /* Map temperature values to Rockchip cores
+          *
+          *   littlecore -> cores 1..4
+          *   bigcore0   -> cores 5,6
+          *   bigcore1   -> cores 7,8
+          */
+         if (existingCPUs == 8 && String_eq(chip->prefix, "littlecore_thermal")) {
```

Comment:
If the first condition is checking for 8 CPUs anyway, why not pull this as an outer layer. That way this automatically groups the three blocks belonging to the Rockchip cores as a side effect.